### PR TITLE
Changes in libevhtp for http support on windows

### PR DESCRIPTION
**Pull request by 9cvele3**
Related to https://github.com/triton-inference-server/server/issues/3130

### CODE REVIEW

**Review by deadeyegoodwin — DISMISSED** [review]

File: libevhtp/libevhtp/evhtp.c [hunk 1]
```diff
@@ -2035,9 +2047,11 @@ htp__request_parse_fini_(htparser * p)
         * callback when selecting the thread to handle future requests. Detail:
         * https://github.com/criticalstack/libevhtp/issues/146#issuecomment-696339470
         */
+       #ifndef EVHTP_DISABLE_EVTHR
```
Comment (GuanLuo) [inline libevhtp/libevhtp/evhtp.c:2050]: `ifdef` macro don't need to have indent

**Review by GuanLuo — COMMENTED** [review]
The change looks good to me. One thing is that we are keeping the third party source unchanged, and have separate [patch files](https://github.com/triton-inference-server/third_party/tree/main/libevhtp) to apply them during Triton build, can you follow the same convention?

**Comment by 9cvele3** [comment]
> One thing is that we are keeping the third party source unchanged, and have separate [patch files](https://github.com/triton-inference-server/third_party/tree/main/libevhtp) to apply them during Triton build,

I need some clarification about this. 
By analyzing some of the patches, it seems to me that the patches are already applied to the source. By applying them again, I get no change. 

For example, in [thread.h.patch](https://github.com/triton-inference-server/third_party/blob/main/libevhtp/thread.h.patch), function `evthr_set_busy` is added. However, [thread.h](https://github.com/triton-inference-server/third_party/blob/main/libevhtp/libevhtp/include/evhtp/thread.h) already contains that change. 

Are the thirdparty sources indeed left unchanged ?

**Comment by GuanLuo** [comment]
My bad for not verifying the file content before making my comment, you are right that the source in this repo is already patched. So what is needed is actually to update the patch files to reflect what is changed comparing the source in this repo with the original source that we checked out from.

**Review by GuanLuo — APPROVED** [review]

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**